### DISCUSSION TITLE
Adding Electricity Consumed sensor on Hitachi model

### DISCUSSION THREAD

**Discussion by rhpijnacker** [Arnold-n/P1P2MQTT#86] [Q&A]
Hi Arnold and community,

I have the P1P2Serial running for a couple of months now on my Hitachi Yutaki S Combi heath-pump.
It is happily producing a number of sensor values on the MQTT interface. Thanks to @Arnold-n for providing this great solution. However the Electricity Consumed is not one of them.

I'm considering investing some time to see if I can add a sensor for this. However, currently I am clueless where to start.
Maybe someone else with a Hitachi heat-pump already made some efforts into this direction?
Otherwise, I would welcome some pointers on getting started, if only on how to build the Hitachi version and upload that to the device.

Thanks,

Ronald

**Comment by Arnold-n** [comment]
Hi @rhpijnacker, thanks for the feedback! Do you know if the Hitachi measures and reports electricity consumption or power usage?

For Daikin there is only low-resolution kWh counter information or inaccurate current reporting, so I plan to support external electricity meters. The Homewizard SMD230-WiFi meter provides great real-time active power info, and support for other Modbus/S0 meters is also planned.

But it would be nicer if the heat pump itself would provide the real-time power consumption. If you or anyone else knows where the power/consumption information is in the H-link2 data (MQTT shows more entities than HA), I can easily provide an updated firmware.

**Comment by rhpijnacker** [comment]
Hi Arnold,

The modbus module for the Hitachi is capable of reporting kWh, so I suppose this is published on the H-Link bus.
My heat-pump is installed on a 3-phase group, so I do think that HomeWizard meter can be used, right?
The outdoor unit is installed on a separate 1-phase group, so that would require a separate meter, is that what you were referring to?

**Comment by Arnold-n** [comment]
Hi Ronald,
Let's try to find the info back in the data!

Either
1. the kWh info is requested via H-link2 when you request it via Modbus, or
2. the kWh info is reported regularly

In case 1, can you log P1P2/R/# during and shortly after a request, preferably in situations where the kWh counter it different; in case 2, can you log P1P2/R/# for a longer time in which the kWh value changes?

Do you know the resolution of the kWh counter?

The SMD230-WiFi is 1-phase, but there is also SDM630-WiFi which is 3-phase. Its resolution is very good, 0.001W and 0.001kWh. Is the inside unit 3-phase? In Daikin systems the compressor is usually located outside.

**Comment by rhpijnacker** [comment]
Hi Arnold,

Unfortunately, I do not actually have a Modbus module connected to the heat-pump (what point would the P1P2Serial interface serve if I had one).
Also I know very little about whether the kWh info is being reported or at what resolution.
I'm happy to log and report any information that would make figuring this out possible, though.

Regarding the heat-pump: the inside unit is 3-phase, the outside unit 1-phase.
